tests/kafkatest/services/console_consumer.py

```diff
@@ -226,10 +230,13 @@ def _worker(self, idx, node):
 
             for line in itertools.chain([first_line], consumer_output):
                 msg = line.strip()
-                if self.message_validator is not None:
-                    msg = self.message_validator(msg)
-                if msg is not None:
-                    self.messages_consumed[idx].append(msg)
+                if msg == "shutdown_complete":
+                    self.nodes_clean_shutdown.append(idx)
+                else:
+                    if self.message_validator is not None:
+                        msg = self.message_validator(msg)
+                    if msg is not None:
+                        self.messages_consumed[idx].append(msg)
```

Comment:
Hmmm I am a bit worried since we are printing out messages if the printing itself can come out of order. Since we are not flushing after every print. Not sure how rare that behavior is...